src/commands/deploy/deploy.ts

```diff
@@ -234,16 +235,15 @@ async function deploy(actionContext: IActionContext, arg1: vscode.Uri | string |
             await updateWorkerProcessTo64BitIfRequired(context, siteConfig, site, language, durableStorageType);
         }
 
-        // app settings shouldn't be checked with flex consumption plans
-        if (isZipDeploy && !isFlexConsumption) {
```

Comment:
What is the likelihood that removing the `!isFlexConsumption` check could introduce breaks to other project deployments?

Reply:
Always, but I wrapped the checks with an if/else clause inside `verifyAppSettings` that should preserve the old behavior.